pmlc/dialect/pxa/transforms/subgroups.cc

```diff
@@ -165,16 +166,28 @@ struct SubgroupCostModel {
   };
 
   // Compute the memory info for a single load/store given the current plan
-  MemInfo computeMemoryInfo(StrideInfo si) {
+  MemInfo computeMemoryInfo(SmallVectorImpl<StrideInfo> &blah) {
     MemInfo out = {0, 1};
-    for (unsigned i = 0; i < ranges.size(); i++) {
-      auto iv = op.getIVs()[i];
-      if (si.strides.count(iv)) {
-        if (plan.subgroupTile[i] == plan.subgroupSize && si.strides[iv] == 1) {
+    for (auto si : blah) {
+      IVLOG(1, "stride info = " << si);
+      int64_t pos = 0;
+      // TODO: add neg
+      for (auto kvp : si.strides) {
+        auto index = kvp.first.getArgNumber();
```

Comment:
So technically this is a bug, since it's valid to have something like this:

affine.parallel (%i) = (0) to (10)
  affine.parallel (%j) = (0) to (10)
    %0 = pxa.load %A[%i + %j]

Now, both %i and %j will have getArgNumber = 0, but they are on different blocks, and perhaps %j is being considered for tiling but %i is not.  Basically this code presumes a single block nest only.

The better way to do this is to make a tile map: DenseMap<BlockArgument, int64_t>for all tiled indexes.  If an index in si.strides is not in the tile map, we should consider it a tile size of 1 (or more simply, don't add anything to 'pos', which is equivalent).

That said, I don't think this bug is the issue with the slow runtimes, since when we call the subgroup pass in the existing pipeline we have a block depth of exactly one since it's basically fresh from the tile lowering.